proposals/NNNN-improve-starting-streaming-services.md

```diff
@@ -0,0 +1,54 @@
+
+The service can fail for many different reasons. Possible failure reasons can be
+- a missing permission due to missing policy update,
+- invalid certificates pending to be updated or
+- discrepancy in clock time causing handshake to fail.
+
+With the procedure of starting services being synchronized and the ability to reset SSL connections, a retry mechanism can be established to improve the user experience launching apps especially for mobile navigation.
+
+## Proposed solution
+
+The solution is to synchronize attempts to start services so that only one starting service is processing. This is essential for services which require encryption/protection as those initiate a TLS handshake on the app's transport session. If the library should start a protected service, it has to ensure the responsible security manager is started. If the attempt to establish a protected service failed due to a `StartServiceNAK`, the library must reset the SSL connection of the security manager instance. This can be done by stopping the security manager. It should be automatically restarted upon the next `StartService` request. This ensures a fluent operation of the security manager across any protected service (including RPC). The next request must not be send within two seconds after the previous attempt failed to avoid unnecessary load for the IVI.
+
+If the app enters a streamable state (reentered `HMI_FULL` and/or `OnHMIStatus.videoStreamingState` is `STREAMABLE`) the libraries start the video manager first followed by the audio manager. If the video manager received NAK for the video service, the library should not start the audio manager's audio service. Instead the library should retry the video service after two seconds of waiting and hold back the audio manager from starting. No other `StarService` request should be send during this time. the infinite retry method is justified due to the app being in a streamable state (HMI_FULL etc.). No other application would be interfering during this time unless the app leaves the streamable state. 
+
+The library should retry until the app leaves the streamable state. If the app left the streamable state during the two seconds, the library must prevent sending the `StartService` request and stop the retry process.
+
+## Potential downsides
+
+In certain situations where the retry mechanism occurs, the proposed solution can increase the time until the app has started the video and audio stream for up to two seconds. However, the libraries gain a lot of stability for error cases which is much more valuable.
+
+It can be possible that this retry mechanism puts some stress to existing infotainment systems as they were never tested against this method. However, the current behavior of the libraries to stop after one failure would leave the user in an unacceptable state with an non-functional app. Therefore a retry mechanism can only improve the user experience.
```

Comment:
```suggestion
It can be possible that this retry mechanism puts some stress on existing infotainment systems as they were never tested against this method. However, the current behavior of the app libraries to stop after one failure would leave the user in an unacceptable state with a non-functional app. Therefore a retry mechanism can only improve the user experience.
```